### PR TITLE
Migrate tests of `server.aggregate` package to JUnit 5 structure

### PR DESCRIPTION
**Pull request by dmitrykuzmin**
This PR starts migration of the `server` module to the new JUnit 5 test structure.

Migration includes adding `@DisplayName` to test cases and test classes, changing naming pattern of tests and also introduction of the `@Nested` test classes for the better structure, as well as some number of lesser changes.

Along with the `server.aggregate` package, the two small test packages, `core` and `envelope`, located under the `server` module, were migrated to the new structure.

### CODE REVIEW

**Comment by dmitrykuzmin** [comment]
@armiol PTAL.

File: server/src/test/java/io/spine/envelope/CommandEnvelopeTest.java [hunk 1]
```diff
@@ -25,36 +25,22 @@
...
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertSame;
 
 /**
  * @author Alexander Yevsyukov
  */
-public class CommandEnvelopeShould
-        extends MessageEnvelopeTest<Command, CommandEnvelope, CommandClass> {
+@SuppressWarnings("DuplicateStringLiteralInspection")
+// Similar test cases to RejectionEnvelopeTest.
```
Comment (armiol) [inline server/src/test/java/io/spine/envelope/CommandEnvelopeTest.java]: Please move the comment to the previous line.

File: server/src/test/java/io/spine/server/aggregate/AggregateMessageDeliveryTest.java [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+    @Nested
+    @DisplayName("in multithreaded env, dispatch commands")
+    class DispatchCommands {
+
+        @Test
+        @DisplayName("to single shard")
+        void toSingleShard() throws Exception {
+            final ParallelDispatcher<ProjectId, Command> dispatcher =
+                    new ParallelDispatcher<ProjectId, Command>(
+                            42, 400, dispatchWaitTime()) {
```
Comment (armiol) [inline server/src/test/java/io/spine/server/aggregate/AggregateMessageDeliveryTest.java]: I think the arguments may fit the previous line.

File: server/src/test/java/io/spine/server/aggregate/AggregateMessageDeliveryTest.java [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+
+            dispatcher.dispatchMessagesTo(new SingleShardProjectRepository());
+        }
+
+        @Test
+        @DisplayName("to several shards")
+        void toSeveralShards() throws Exception {
+            final ParallelDispatcher<ProjectId, Command> dispatcher =
+                    new ParallelDispatcher<ProjectId, Command>(
+                            23, 423, dispatchWaitTime()) {
```
Comment (armiol) [inline server/src/test/java/io/spine/server/aggregate/AggregateMessageDeliveryTest.java]: Same as above.

File: server/src/test/java/io/spine/server/aggregate/AggregateMessageDeliveryTest.java [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+    @Nested
+    @DisplayName("in multithreaded env, dispatch events")
+    class DispatchEvents {
+
+        @Test
+        @DisplayName("to single shard")
+        void toSingleShard() throws Exception {
+            final ParallelDispatcher<ProjectId, Event> dispatcher =
+                    new ParallelDispatcher<ProjectId, Event>(
+                            130, 500, dispatchWaitTime()) {
```
Comment (armiol) [inline server/src/test/java/io/spine/server/aggregate/AggregateMessageDeliveryTest.java]: Same as above.

File: server/src/test/java/io/spine/server/aggregate/AggregateMessageDeliveryTest.java [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+    @Nested
+    @DisplayName("in multithreaded env, dispatch rejections")
+    class DispatchRejections {
+
+        @Test
+        @DisplayName("to single shard")
+        void toSingleShard() throws Exception {
+            final ParallelDispatcher<ProjectId, Rejection> dispatcher =
+                    new ParallelDispatcher<ProjectId, Rejection>(
+                            36, 12, dispatchWaitTime()) {
```
Comment (armiol) [inline server/src/test/java/io/spine/server/aggregate/AggregateMessageDeliveryTest.java]: Once more, please try to fit the previous line.

File: server/src/test/java/io/spine/server/aggregate/AggregatePartRepositoryLookupTest.java [hunk 6]
```diff
@@ -72,27 +71,27 @@ public void find_a_repository() {
     }
 
     @Test
-    public void throw_if_repository_is_not_AggregatePartRepository() {
+    @DisplayName("throw if repository not found")
```
Comment (armiol) [inline server/src/test/java/io/spine/server/aggregate/AggregatePartRepositoryLookupTest.java]: `isn't found`

File: server/src/test/java/io/spine/server/aggregate/AggregatePartRepositoryLookupTest.java [hunk 7]
```diff
@@ -72,27 +71,27 @@ public void find_a_repository() {
...
-                createLookup(boundedContext, TaskId.class, Task.class);
-        thrown.expect(IllegalStateException.class);
-        lookup.find();
+                createLookup(boundedContext, Timestamp.class, StringValue.class);
+        assertThrows(IllegalStateException.class, lookup::find);
     }
 
     @Test
-    public void throw_if_repository_not_found() {
+    @DisplayName("throw if found repository is not AggregatePartRepository")
```
Comment (armiol) [inline server/src/test/java/io/spine/server/aggregate/AggregatePartRepositoryLookupTest.java]: Throw what?
Also, `if the repo found is not AggregatePartRepository`.

File: server/src/test/java/io/spine/server/aggregate/AggregatePartRepositoryLookupTest.java [hunk 8]
```diff
@@ -72,27 +71,27 @@ public void find_a_repository() {
...
-                createLookup(boundedContext, Timestamp.class, StringValue.class);
-        thrown.expect(IllegalStateException.class);
-        lookup.find();
+                createLookup(boundedContext, TaskId.class, Task.class);
+        assertThrows(IllegalStateException.class, lookup::find);
     }
 
     @Test
-    public void throw_if_id_class_does_not_match() {
+    @DisplayName("throw if found repository id class does not match root id class")
```
Comment (armiol) [inline server/src/test/java/io/spine/server/aggregate/AggregatePartRepositoryLookupTest.java]: Similar to the one above. Throw what?

`if the ID class of the repo found does not match the ID class of its root`.

File: server/src/test/java/io/spine/server/aggregate/AggregateRepositoryTest.java [hunk 9]
```diff
@@ -0,0 +1,583 @@
...
+            return optional.get();
+        }
+    }
+
+    @Nested
+    @DisplayName("manage snapshots and event count properly")
+    class ManageSnapshots {
+
+        @Test
+        @DisplayName("when required to store snapshot")
```
Comment (armiol) [inline server/src/test/java/io/spine/server/aggregate/AggregateRepositoryTest.java]: `it's required to store a snapshot`

File: server/src/test/java/io/spine/server/aggregate/AggregateRepositoryTest.java [hunk 10]
```diff
@@ -0,0 +1,583 @@
...
+
+            repository.store(aggregate);
+            AggregateStateRecord record = readRecord(aggregate);
+            assertTrue(record.hasSnapshot());
+            assertEquals(0, repository.aggregateStorage()
+                                      .readEventCountAfterLastSnapshot(aggregate.getId()));
+        }
+
+        @Test
+        @DisplayName("when storing snapshot not needed")
```
Comment (armiol) [inline server/src/test/java/io/spine/server/aggregate/AggregateRepositoryTest.java]: `isn't needed`

File: server/src/test/java/io/spine/server/aggregate/AggregateRepositoryTest.java [hunk 11]
```diff
@@ -0,0 +1,583 @@
...
+
+            repository.store(aggregate);
+
+            assertTrue(repository.find(aggregate.getId())
+                                 .isPresent());
+        }
+    }
+
+    @Test
+    @DisplayName("not create new aggregates on find")
```
Comment (armiol) [inline server/src/test/java/io/spine/server/aggregate/AggregateRepositoryTest.java]: `upon lookup`

File: server/src/test/java/io/spine/server/aggregate/AggregateRepositoryTest.java [hunk 12]
```diff
@@ -0,0 +1,583 @@
...
+@DisplayName("AggregateRepository should")
+public class AggregateRepositoryTest {
+
+    private final TestActorRequestFactory requestFactory =
+            TestActorRequestFactory.newInstance(getClass());
+
+    private BoundedContext boundedContext;
+    private AggregateRepository<ProjectId, ProjectAggregate> repository;
+
+    private static ProjectId givenAggregateId(String id) {
```
Comment (armiol) [inline server/src/test/java/io/spine/server/aggregate/AggregateRepositoryTest.java]: Can we please move the bounded context, the repository, the request factory + `given...` methods to the `...TestEnv` class? Then expose those via the test env API (static methods, e.g. `boundedContext()`, `given...` etc).

File: server/src/test/java/io/spine/server/aggregate/AggregateTest.java [hunk 13]
```diff
@@ -0,0 +1,966 @@
...
+            ModelTests.clearModel();
+            AggregateWithMissingApplier aggregate = new AggregateWithMissingApplier(ID);
+
+            // Pass a command for which the target aggregate does not have a handling method.
+            assertThrows(IllegalStateException.class,
+                         () -> dispatchCommand(aggregate, env(addTask)));
+        }
+
+        @Test
+        @DisplayName("event applier for non state-neutral event")
```
Comment (armiol) [inline server/src/test/java/io/spine/server/aggregate/AggregateTest.java]: `event applier for the event emitted in a result of command handling`

File: server/src/test/java/io/spine/server/aggregate/EventApplierMethodTest.java [hunk 14]
```diff
@@ -81,72 +91,99 @@ public void create_method() {
...
-    @Test
-    public void consider_not_private_applier_valid() {
-        final Method method = new ValidApplierButNotPackagePrivate().getMethod();
+            assertIsEventApplier(applier);
+        }
 
-        assertIsEventApplier(method);
-    }
+        @Test
+        @DisplayName("when it's not private")
```
Comment (armiol) [inline server/src/test/java/io/spine/server/aggregate/EventApplierMethodTest.java]: I think `when` is a common part, that may be extracted to the parent display name.

File: server/src/test/java/io/spine/server/aggregate/EventApplierMethodTest.java [hunk 15]
```diff
@@ -81,72 +91,99 @@ public void create_method() {
...
+            final Method applier = new InvalidApplierOneNotMsgParam().getMethod();
 
-    private static void assertIsEventApplier(Method applier) {
-        assertTrue(EventApplierMethod.predicate().apply(applier));
-    }
+            assertIsNotEventApplier(applier);
+        }
+
+        @Test
+        @DisplayName("when it has non-void return type")
```
Comment (armiol) [inline server/src/test/java/io/spine/server/aggregate/EventApplierMethodTest.java]: Same as above. `when` may be extracted.

File: server/src/test/java/io/spine/server/aggregate/IdempotencyGuardTest.java [hunk 16]
```diff
@@ -80,11 +84,12 @@ public void throw_DuplicateCommandException_when_the_command_was_handled_since_l
 
         final IgTestAggregate aggregate = repository.loadAggregate(tenantId, projectId);
         final IdempotencyGuard guard = new IdempotencyGuard(aggregate);
-        guard.check(of(createCommand));
+        assertThrows(DuplicateCommandException.class, () -> guard.check(of(createCommand)));
     }
 
     @Test
-    public void not_throw_when_the_command_was_handled_but_the_snapshot_was_made() {
+    @DisplayName("not throw when command was handled but snapshot was made")
```
Comment (armiol) [inline server/src/test/java/io/spine/server/aggregate/IdempotencyGuardTest.java]: Throw what? Same below.

**Comment by dmitrykuzmin** [comment]
@armiol PTAL.

**Pull request merged**